### PR TITLE
Add comments in user config struct

### PR DESCRIPTION
**Pull request by jesseduffield**
We're going to make this user config struct a more authoritative source of truth.

Some of these fields weren't actually explained anywhere so I've added explanations.

In places where a lot of explanation is required I've linked to existing explanations in other docs.

- **PR Description**

- **Please check if the PR fulfills these requirements**

* [ ] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [ ] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [ ] Docs (specifically `docs/Config.md`) have been updated if necessary
* [ ] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -20,7 +20,8 @@ linters:
 linters-settings:
   exhaustive:
     default-signifies-exhaustive: true
-
+  staticcheck:
+    checks: ["all", "-SA1019"]
```
Comment (stefanhaller) [inline .golangci.yml:24]: What's this? Did you include this on purpose? If so, might be useful to make this a separate commit and explain why this is added.

**Review by stefanhaller — COMMENTED** [review]
This is great. I only skimmed the comments, don't have time to proof-read everything, but I'm sure they are good.

I think we should merge this soon because it conflicts with other changes to UserConfig.

File: .golangci.yml (hunk 1 above)
Comment (jesseduffield) [reply .golangci.yml:27]: Ah yes. See https://github.com/jesseduffield/lazygit/pull/3040/commits/3e2ef84d566678ffaa6afa120c6881b558eb2227

Comment (stefanhaller) [reply .golangci.yml:27]: Nice, thanks.

Comment (jesseduffield) [reply .golangci.yml:27]: Ah yes. See https://github.com/jesseduffield/lazygit/pull/3040/commits/3e2ef84d566678ffaa6afa120c6881b558eb2227
